src/devices/Tlc1543/Tlc1543.cs

```diff
@@ -0,0 +1,239 @@
+        private readonly int _inputOutputClock;
+        private readonly int? _endOfConversion;
+        private readonly int _channels = 11;
+        private readonly TimeSpan _conversionTime = new TimeSpan(210);
+        private GpioController _digital;
+        private bool _shouldDispose;
+        private Channel _chargeChannel = Channel.SelfTest512;
+
+        /// <summary>
+        /// Channel used between readings to charge up capacitors in ADC
+        /// <remarks>
+        /// <br>Refer to table 2 and 3 in TLC1543 documentation for more information</br>
+        /// </remarks>
+        /// </summary>
+        public Channel ChargeChannel
+        {
+            get => _chargeChannel;
+            set
+            {
+                if ((int)value >= 0 & (int)value < 14)
+                {
+                    _chargeChannel = value;
+                }
+                else
+                {
+                    throw new ArgumentOutOfRangeException("ChargeChannel", "Out of range. \nMust be non-negative and less than 14. \nRefer to table 2 and 3 in TLC1543 documentation for more information");
+                }
+            }
+        }
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Constructor for Tlc1543
+        /// </summary>
+        /// <param name="address">Address</param>
+        /// <param name="chipSelect">Chip Select</param>
+        /// <param name="dataOut">Data Out</param>
+        /// <param name="inputOutputClock">I/O Clock</param>
+        /// <param name="controller"> s</param>
+        /// <param name="pinNumberingScheme"> s</param>
+        /// <param name="shouldDispose"> s</param>
+        public Tlc1543(int address, int chipSelect, int dataOut, int inputOutputClock, GpioController controller = null, PinNumberingScheme pinNumberingScheme = PinNumberingScheme.Logical, bool shouldDispose = true)
+        {
+            _shouldDispose = controller == null || shouldDispose;
+            _digital = controller ?? new GpioController(pinNumberingScheme);
+            _address = address;
+            _chipSelect = chipSelect;
+            _dataOut = dataOut;
+            _inputOutputClock = inputOutputClock;
+
+            _digital.OpenPin(_address, PinMode.Output);
+            _digital.OpenPin(_chipSelect, PinMode.Output);
+            _digital.OpenPin(_dataOut, PinMode.InputPullUp);
+            _digital.OpenPin(_inputOutputClock, PinMode.Output);
+        }
+
+        /// <summary>
+        /// Constructor for Tlc1543
+        /// </summary>
+        /// <param name="address">Address</param>
+        /// <param name="chipSelect">Chip Select</param>
+        /// <param name="dataOut">Data Out</param>
+        /// <param name="endOfConversion">End of Conversion</param>
+        /// <param name="inputOutputClock">I/O Clock</param>
+        public Tlc1543(int address, int chipSelect, int dataOut, int inputOutputClock, int endOfConversion)

```

Comment:
is this using SPI? if so please use SpiDevice (you can use SoftwareSpi for manual pins)

Reply:
(see sample in Mcp3008 for how to do that)

Reply:
As have been said before let's have it with GPIO support only for now and consider SPI in future. 
And from what I have been testing in last few days I've noticed many problems in using SoftwareSpi as well as SpiDevice with this device (Interrupts while clocking or simply inconsistencies while reading data)

Example of such interrupt:
![](https://user-images.githubusercontent.com/24566138/95649621-a14b0d80-0ade-11eb-8cfc-b03d8340386f.jpg)

Reply:
If you are using the hardware SPI, it should just work as expected. For the software Spi, it depends of the frequency used and also you should make sure you have the precision timer. If you have precision timer, that should just work as well.
What were the parameters you used for this graph? (hardware or software, frequency, precision timer, etc)

Reply:
SoftwareSpi
```cs
SpiConnectionSettings connectionSettingsSoft = new SpiConnectionSettings(0)
            {
                ClockFrequency = 2_100_000,
                Mode = SpiMode.Mode0,
                DataFlow = DataFlow.MsbFirst,
                ChipSelectLineActiveState = PinValue.Low,
            };
```
I have tested that so many times already and it's so inconsistent between runs that i got discouraged from implementing it further. HardwareSpi works good for what it's worth but that has its share of problems (not able to go for less than 16 clock cycle or using End Of Conversion pin for power/time saving without using both spi and gpio

Reply:
I understand that software SPI is not working and that hardware SPI is. So I would then add an SPI interface and mentioning that only hardware SPI is proven to work, not software SPI.
And adding some comments that hardware SPI may not be as efficient as GPIO. Now, some may prefer to use SPI over GPIO for different reasons and may not be too stressed about the few extra clocks and timing.

Reply:
Yes, I plan to add that later on. For now i want to have working module with GPIO, then implement rest of the 6 modes that the ADC supports (and proper EOC handling), next limited hardware SPI and some kind of software SPI (I may look into improving software SPI binding at that point too, or it may be improved already at this point in future)

Reply:
BTW you can go for less than 16 cycles, change DataBitLength to 7 and send 2 bytes, that way you will go exactly 14 bits. I'm not in favor of adding just GPIO support because GPIO is already supported with SoftwareSpiDevice

Reply:
See code: https://github.com/dotnet/iot/blob/master/src/devices/SoftwareSpi/SoftwareSpi.cs#L160